ui/src/views/WorkerManager/WMEditWorkerPool/index.jsx

```diff
@@ -0,0 +1,203 @@
+import { hot } from 'react-hot-loader';
+import React, { Component } from 'react';
+import { bool } from 'prop-types';
+import TextField from '@material-ui/core/TextField/TextField';
+import FormGroup from '@material-ui/core/FormGroup/FormGroup';
+import FormControlLabel from '@material-ui/core/FormControlLabel/FormControlLabel';
+import FormLabel from '@material-ui/core/FormLabel/FormLabel';
+import MenuItem from '@material-ui/core/MenuItem/MenuItem';
+import CodeEditor from '@mozilla-frontend-infra/components/CodeEditor';
+import ListItem from '@material-ui/core/ListItem/ListItem';
+import Switch from '@material-ui/core/Switch/Switch';
+import { withStyles } from '@material-ui/core';
+import CheckIcon from 'mdi-react/CheckIcon';
+import List from '../../Documentation/components/List';
+import isWorkerTypeNameValid from '../../../utils/isWorkerTypeNameValid';
+import Dashboard from '../../../components/Dashboard';
+import Button from '../../../components/Button';
+
+const gcpConfig = {
+  minCapacity: 0,
+  maxCapacity: 0,
+  capacityPerInstance: 1,
+  machineType: 'n1-highcpu-8',
+  regions: ['us-west2'],
+  userData: {},
+  scheduling: {},
+  networkInterfaces: [{}],
+  disks: [{}],
+};
+
+@hot(module)
+@withStyles(theme => ({
+  successIcon: {
+    ...theme.mixins.successIcon,
+  },
+  createIconSpan: {
+    ...theme.mixins.fab,
+    ...theme.mixins.actionButton,
+    right: theme.spacing.unit * 11,
+  },
+}))
+export default class WMWorkerPoolEditor extends Component {
+  static defaultProps = {
+    isNewWorkerPool: true,
+  };
+
+  static propTypes = {
+    isNewWorkerPool: bool,
+  };
+
+  state = {
+    workerPool: {
+      name: '',
+      description: '',
+      owner: '',
+      wantsEmail: false,
+      providerType: 'gcp',
```

Comment:
Nit:

```jsx
providers = ['GCP'];

// ...

state = {
  // GCP
  providerType: providers[0],
};

// ...

<TextField // ...>
  {providers.map(provider => <MenuItem key={provider} value={provider}>{provider}</MenuItem>)}
</TextField>
```

Feel free to ignore this if you don't feel it's necessary right now.

Reply:
Yeah, I can use an enum kind of thing